dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/map-to-premium-payments-groups.ts

```diff
@@ -0,0 +1,50 @@
+import { calculateTotalCurrencyAndAmount, getFeeRecordPaymentEntityGroupStatus } from '../../../../../helpers';
+import { mapPaymentEntityToPayment } from '../../../../../mapping/payment-mapper';
+import { mapFeeRecordEntityToFeeRecord } from '../../../../../mapping/fee-record-mapper';
+import { PremiumPaymentsGroup } from '../../../../../types/utilisation-reports';
+import { FeeRecordPaymentEntityGroup } from '../../../../../types/fee-record-payment-entity-group';
+
+/**
+ * Maps the fee record payment entity groups to the premium payments groups
+ * @param feeRecordPaymentEntityGroups - The fee record payment entity groups
+ * @returns The premium payments groups
+ * @throws {Error} If the group has no payments but has multiple fee records
+ */
+export const mapToPremiumPaymentsGroups = (feeRecordPaymentEntityGroups: FeeRecordPaymentEntityGroup[]): PremiumPaymentsGroup[] =>
+  feeRecordPaymentEntityGroups.map((group) => {
+    const groupStatus = getFeeRecordPaymentEntityGroupStatus(group);
+
+    if (group.payments.length === 0) {
+      // If there are no payments, there is only one fee record in the group
```

Comment:
Could/should we move the proceeding logic into it's own function? Would make `mapToPremiumPaymentsGroups` simpler

Reply:
Split this logic out into two sub cases:
- `mapGroupWithNoPaymentsToPremiumPaymentsGroup`
- `mapGroupWithPaymentsToPremiumPaymentsGroup`
The main mapping function is now a lot simpler, thanks for the suggestion